pkg/apis/imageregistry/v1alpha1/imageregistry_types.go

```diff
@@ -0,0 +1,106 @@
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ImageRegistrySpec is the spec of managedClusterImageRegistry.
+type ImageRegistrySpec struct {
+	// Registry is the address of overridden image registry
+	// +kubebuilder:validation:Required
+	// +required
+	Registry string `json:"registry"`
+
+	// PullSecret is the name of image pull secret which should be in the same namespace with the managedClusterImageRegistry.
+	// +kubebuilder:validation:Required
+	// +required
+	PullSecret corev1.LocalObjectReference `json:"pullSecret"`
+
+	// PlacementRef is the referred Placement name.
+	// +kubebuilder:validation:Required
+	// +required
+	PlacementRef PlacementRef `json:"placementRef"`
+}
+
+// PlacementRef is the referred placement
+type PlacementRef struct {
+	// Group is the api group of the placement. Current group is cluster.open-cluster-management.io.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Enum=cluster.open-cluster-management.io
+	// +required
+	Group string `json:"group"`
+
+	// Resource is the resource type of the Placement. Current resource is placement or placements.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Enum=Placement;Placements
```

Comment:
I think it should be placement;placements.

Reply:
done